yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1264,7 +1266,7 @@ object Client extends Logging {
 
       val secondaryJars =
         if (args != null) {
-          getSecondaryJarUris(Option(args.addJars).map(_.split(",").toSeq))
+          getSecondaryJarUris(sparkConf.get(JARS_TO_DISTRIBUTE).map(_.split(",").toSeq))
```

Comment:
note: if you pre-process `JARS_TO_DISTRIBUTE`  with `resolveURIs` you should probably update `sparkConf` so that this code gets the pre-processed values too.